File: Leetcode/27-Remove-Element/RemoveElement.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    if (!nums || nums.length == 0) {
        return 0
    }
    var i=0
    var j=0
    while (j<nums.length) {
        if (nums[j]!=val) {
            nums[i] = nums[j]
            i++
        }
        j++
    }
    return i
};

File: Leetcode/27-Remove-Element/RemoveElement.go
func removeElement(nums []int, val int) int {
	if nums == nil || len(nums) == 0 {
		return 0
	}
	i := 0
	j := 0
	length := len(nums)
	for j < length {
		if nums[j] != val {
			nums[i] = nums[j]
			i++
		}
		j++
	}
	return i
}
